### DISCUSSION TITLE
Run Redash behind some reverse proxy

### DISCUSSION THREAD

**Discussion by elluvium** [getredash/redash#5953] [Q&A]
Hi there! 
I did not find answers to my questions in the official docs so decided to ask here. Basically, we're trying to move our Redash instance behind Kong and struggling here with weird behavior: Redash instance sets the RST flag to kong's attempt to establish a connection and  Kong throws
`upstream connect error or disconnect/reset before headers. reset reason: connection failure`
error.

tcpdump shows that traffic successfully reaches Redash but:

- Redash does not write any useful information to logs about this RST flags even with debug log level
- A direct route to Redash pod works fine without RST flags

So I am a little bit stuck and have two questions:
1. Does Redash able to run behind reverse proxy/kong/nginx at all?
2. Is there any tricks available to make it work?

**Comment by justinclift** [comment]
> 1. Does Redash able to run behind reverse proxy/kong/nginx at all?

I'm not familiar with Kong at all, however the fairly established Docker deployment approach for Redash does proxy through nginx.  Looking over the config used for that *might* help a bit?

**Comment by majormoses** [comment]
We have used https://github.com/oauth2-proxy/oauth2-proxy successfully
